src/libstore/local-store.cc

```diff
@@ -1595,28 +1598,23 @@ bool LocalStore::verifyStore(bool checkContents, RepairFlag repair)
 }
 
 
-void LocalStore::verifyPath(const Path & pathS, const StringSet & store,
-    PathSet & done, StorePathSet & validPaths, RepairFlag repair, bool & errors)
+void LocalStore::verifyPath(const StorePath & path, std::function<bool(const StorePath &)> existsInStoreDir,
```

Comment:
Can't this lambda be avoided by changing the `store` argument to a `StorePathSet`?

Reply:
It will need to be a lambda in [this pull request](https://github.com/NixLayeredStore/nix/pull/13) so we can change the behaviour for `LocalOverlayStore`, but it doesn't need to be yet so I've changed it back to a set.

Reply:
Yes oops I had meant for the lambda change to not be part of this.